### PR TITLE
[develop] Carlos - Count most frequent item

### PR DESCRIPTION
**Pull request by carlosmc23**

### CODE REVIEW

File: src/main/java/org/fundacionjala/coding/carlos/countfrequentitem/CountFrequentItem.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class CountFrequentItem {
+
+    /**
+     * @param arrayNumbers to find the most frequent item
+     * @return the number of times that the most frequent item is repeated in an array.
+     */
+    public int getMostFrequent(final int[] arrayNumbers) {
+        int number = 0;
+        int repeated = 0;
+        for (int arrayNumber : arrayNumbers) {
```
Comment (marcos-x86) [inline src/main/java/org/fundacionjala/coding/carlos/countfrequentitem/CountFrequentItem.java]: Please consider removing duplicates from the original array first, according to your solution every number in the array will compute `verifyFrequency()` method including numbers already processed (the duplicated ones). Consider removing duplicates to improve performance with:
```java
LinkedHashSet<Integer> linkedHashSet = new LinkedHashSet<>(Arrays.asList(arrayNumbers));
Integer[] numbersWithoutDuplicates = linkedHashSet.toArray(new Integer[] {});
```

File: src/main/java/org/fundacionjala/coding/carlos/countfrequentitem/CountFrequentItem.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            number = arrayNumber;
+            if (repeated < verifyFrequency(number, arrayNumbers)) {
```
Comment (marcos-x86) [inline src/main/java/org/fundacionjala/coding/carlos/countfrequentitem/CountFrequentItem.java]: You are executing `verifyFrequency()` method two times, please consider store the result in a variable to reuse the value.
```java
int frequency = verifyFrequency(number, arrayNumbers);
if(repeated < frequency) {
  repeated = frecuency;
}
```

**Review by marcos-x86 — CHANGES_REQUESTED** [review]
You are including `CalculateNumbers.java` and `HighestLowestTest.java` in your PR, your solutions were already merged, so please consider removing these two classes from your PR.

**Comment by carlosmc23** [comment]
Hi Marcos I found these ways to solve the exercise

- Set<Integer> linkedHashSet = new LinkedHashSet<>();  your suggestion
- list.stream().distinct().collect(Collectors.toList());
- Collections.frequency(aLista,s)

but I am using this and I think is a better way:

- map.merge(i, 1, Integer::sum);

Where merge is a method that searches if the key exists in your Map, if it does not exist it creates it with the value that you have passed to it (1), and if it exists it applies to the value that it already has a mapping with the value that you pass it ( in this case we sum 1 to the value that was previously associated)

**Review by marcos-x86 — APPROVED** [review]
LGTM!

**Pull request merged**